extensions/replication/queueProcessor/QueueProcessor.js

```diff
@@ -66,8 +67,31 @@ class _UserAuthManager {
 }
 
 class _RoleAuthManager {
-    constructor() {
-        throw Error('Role-based authentication not yet implemented');
+    constructor(authConfig, log) {
+        this._log = log;
+        const { host, port } = authConfig.vault;
+        this._vaultclient = new Vaultclient(host, port);
+    }
+
+    getCredentialProvider() {
```

Comment:
Yeah I am not sure, it was something that I worked with last night. I will make changes as needed.